Src/AutoFakeItEasy.FakeItEasy2UnitTest/AutoFakeItEasy.FakeItEasy2UnitTest.csproj

```diff
@@ -8,6 +8,7 @@
     <AssemblyTitle>AutoFakeItEasy.FakeItEasy2.UnitTest</AssemblyTitle>
     <AssemblyName>AutoFixture.AutoFakeItEasy.FakeItEasy2UnitTest</AssemblyName>
     <RootNamespace>AutoFixture.AutoFakeItEasy.UnitTest</RootNamespace>
+    <DefineConstants>CAN_FAKE_DELEGATES</DefineConstants>
```

Comment:
What is the purpose of `CAN_FAKE_DELEGATES` constant?

I haven't followed this pull request from the beginning, but @zvirja [asked](https://github.com/AutoFixture/AutoFixture/pull/986#issuecomment-371804702) if I could take a look, so hence the question. 😅

Reply:
The minimum supported version of FakeItEasy doesn't support the delegate generation we are trying to enable. However, we added the feature as more recent versions of FakeItEasy support it. 

We have unit tests for different version of the `FakeItEasy` library. The purpose of the constant is to specify whether the delegates feature is supported for the version of FakeItEasy we depend on.

Reply:
@moodmosaic Did you have a chance to look into this? 😊 I'd like to add a few more related changes before the release, so it might take a while before the next release.. 😟

Reply:
@zvirja, I've just quickly gone through the changes. Feel free to move on with this 👍